frontend/src/utilities.ts

```diff
@@ -36,3 +40,30 @@ export const getFromAPI = async <$ReturnType>(url: string, data?: unknown, metho
 export const generateRange = (size: number, start = 0) => {
     return Array.from({ length: size }).map((_, i) => i + start);
 };
+
+/**
+ * Generate string date format from date
+ * @param date Date object or string in format 'YYYY-MM-DD' or 'YYYY-MM-DDTHH:MMZ'
+ * @param withTime Include time in the output
+ * @returns If withTime is false 'YYYY-MM-DD', otherwise 'YYYY-MM-DD HH:MM'
+ */
+export const getDate = (date: Date | string, withTime = false) => {
```

Comment:
Let's use a library for this, e.g. https://date-fns.org/. No need to reimplement these kinds of hairy logic by hand, we want to get rid of similar hacks from Kelvin.